mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +130,35 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    # we don't want to decode times, since we're typically using dates
+    # outside the allowed range
+    kwargs['decode_times'] = False
+    # files differ along the Time dimension
+    kwargs['concat_dim'] = 'Time'
+
+    autoclose = ((autocloseFileCount is not None) and
+                 len(fileNames) > autocloseFileCount)
+
+    openedWithAutoclose = False
+    if autoclose:
+        try:
+            ds = xarray.open_mfdataset(fileNames,
+                                       preprocess=preprocess_partial,
+                                       autoclose=True, **kwargs)
```

Comment:
Dude, @pwolfram, you really can be "judgy" about code.  "This is not intuitive" is kind of harsh.  You mean, I presume, "I would do this differently."

But okay, I will take out kwargs.  I will instead pass in the config.  This will (at some later point) allow us to support dask chunking which seems like something we might already need for the MOC calculations, for example.

I will support `autoclose` internally.  I can use the heuristic you are suggesting, if you give me some code for it...